rs/sns/testing/tests/sns_testing_ci.rs

```diff
@@ -1,29 +1,60 @@
-use candid::{Decode, Encode, Principal};
+use std::path::PathBuf;
+
+use candid::{CandidType, Decode, Encode, Principal};
 use canister_test::Wasm;
 use ic_base_types::{CanisterId, PrincipalId};
 use ic_management_canister_types::CanisterSettings;
+use ic_nervous_system_agent::pocketic_impl::PocketIcAgent;
 use ic_nervous_system_integration_tests::pocket_ic_helpers::{
     install_canister_on_subnet, load_registry_mutations, NnsInstaller, STARTING_CYCLES_PER_CANISTER,
 };
+use ic_nns_common::pb::v1::NeuronId;
 use ic_nns_constants::{LEDGER_INDEX_CANISTER_ID, ROOT_CANISTER_ID};
 use ic_sns_testing::nns_dapp::bootstrap_nns;
-use ic_sns_testing::sns::{
-    pocket_ic::{create_sns, install_test_canister, upgrade_sns_controlled_test_canister},
-    TestCanisterInitArgs,
-};
+use ic_sns_testing::sns::pocket_ic::{create_sns, propose_sns_controlled_canister_upgrade};
+use ic_sns_testing::sns::{await_sns_controlled_canister_upgrade, sns_proposal_upvote};
 use ic_sns_testing::utils::{
     validate_network, validate_target_canister, SnsTestingCanisterValidationError,
     SnsTestingNetworkValidationError, TREASURY_PRINCIPAL_ID,
 };
 use icp_ledger::Tokens;
+use pocket_ic::nonblocking::PocketIc;
 use pocket_ic::PocketIcBuilder;
 use tempfile::TempDir;
 
-#[tokio::test]
-async fn test_sns_testing_basic_scenario() {
+const DEV_PARTICIPANT_ID: PrincipalId = PrincipalId::new_user_test_id(1000);
+const DEV_NEURON_ID: NeuronId = NeuronId { id: 1000u64 };
+
+// TODO @rvem: I don't like the fact that this struct definition is copy-pasted from 'canister/canister.rs'.
+// We should extract it into a separate crate and reuse in both canister and this crates.
```

Comment:
Agree for production canisters (this is our practice we've adopted and been migrating towards), but in the case of test canisters this is often more work than it's worth.

Reply:
This type ended up being used only in sns-testing tests and it's highly unlikely to change, so I agree that it should be fine to keep it as is